addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -71,6 +74,80 @@ def _assert_picking_related_pickings_match_expected_values(self, pickings, expec
 
 
 class TestStockPicking(TestStockPickingCommon):
+    def test_picking_naming_convention(self):
+        """
+        Test that backorders get created with a -001 suffix,
+        but non-backordered pickings are not.
+
+        Create a picking and backorder twice to test it holds for chained
+        pickings.
+        """
+        self.create_quant(self.fig.id, self.test_stock_location_02.id, 50)
+        pick = self.Picking.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{"product": self.fig, "uom_qty": 50}],
+            location_id=self.test_stock_location_02.id,
+        )
+        pick.move_lines[0].quantity_done = 10
+        pick._action_done()
+        bk_picking_1 = pick.u_created_backorder_ids
+        bk_picking_1.move_lines[0].quantity_done = 10
+        bk_picking_1._action_done()
+        bk_picking_2 = bk_picking_1.u_created_backorder_ids
+
+        # Check naming convention
+        self.assertNotRegex("-", pick.name)
+        self.assertEqual(bk_picking_1.name, pick.name + "-001")
+        self.assertEqual(bk_picking_2.name, pick.name + "-002")
+
+    @unittest.skip("Takse too long")
+    def test_picking_naming_convention_exceeds_limit(self):
```

Comment:
Can we replicate the same test but with fewer picks by chaanging the padding to `00` instead of `000`?